activesupport/lib/active_support/evented_file_update_checker.rb

```diff
@@ -55,17 +55,15 @@ def initialize(files, dirs = {}, &block)
       dtw = directories_to_watch
       @dtw, @missing = dtw.partition(&:exist?)
 
-      if @dtw.any?
-        # Loading listen triggers warnings. These are originated by a legit
-        # usage of attr_* macros for private attributes, but adds a lot of noise
-        # to our test suite. Thus, we lazy load it and disable warnings locally.
-        silence_warnings do
-          require "listen"
-        rescue LoadError => e
-          raise LoadError, "Could not load the 'listen' gem. Add `gem 'listen'` to the development group of your Gemfile", e.backtrace
-        end
+      # Loading listen triggers warnings. These are originated by a legit
+      # usage of attr_* macros for private attributes, but adds a lot of noise
+      # to our test suite. Thus, we lazy load it and disable warnings locally.
+      silence_warnings do
```

Comment:
If we are always requiring this file it is better to just require on the top of the file.

Reply:
I think this may make sense to load here. We always require this file, but listen is an optional dependency. We only want an error if a user tries to use `EventedFileUpdateChecker` without listen installed.

``` console
> ActiveSupport::EventedFileUpdateChecker
=> ActiveSupport::EventedFileUpdateChecker
> ActiveSupport::EventedFileUpdateChecker.new(['something.txt']){}
LoadError: Could not load the 'listen' gem. Add `gem 'listen'` to the development group of your Gemfile
from /Users/jhawthorn/.gem/ruby/2.6.3/gems/zeitwerk-2.1.6/lib/zeitwerk/kernel.rb:23:in `require'
Caused by LoadError: cannot load such file -- listen
from /Users/jhawthorn/.gem/ruby/2.6.3/gems/zeitwerk-2.1.6/lib/zeitwerk/kernel.rb:23:in `require'
```

Reply:
I don't think we require this file anywhere. At least I could not find. We always rely on autoload to load this file.

Reply:
🙇 That makes sense

Reply:
Ok so the `listen` gem (and this file) will only be loaded when we call the constant, right ?

Reply:
Right.

Reply:
removed!